src/ow1/reset_hero.opy

```diff
@@ -71,3 +72,14 @@ rule "[reset_hero.opy]: Reset hero on hero switch":
 
     wait()
     resetHero()
+
+
+def respawnSameState():
+    @Name "[reset_hero.opy]: respawnSameState()"
+    eventPlayer.player_health = eventPlayer.getHealth()
+    kill(eventPlayer,null)
+    # Use resurrect to keep the same position and angle
+    eventPlayer.resurrect()
+    # Wait to set healt else roadhog will be full life
```

Comment:
Just fix this little typo and we're set.

Reply:
dont use resurrect

Reply:
Okay what is the reason to not use it ?

Reply:
I was wary of that too but it works in my testing.

Reply:
> I was wary of that too but it works in my testing. Roadhog still takes damage as expected.

Hm ecksdee said you don't take damage until 3 seconds after and I had the same problem. Maybe set health fixes it.

Reply:
> dont use resurrect

The resurrected hog player is immune to damage after rez until he uses an ability. @novalderya